crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -618,6 +605,60 @@ impl EventTimelineItem {
     pub fn local_echo_send_handle(&self) -> Option<SendHandle> {
         as_variant!(self.handle(), TimelineItemHandle::Local(handle) => handle.clone())
     }
+
+    /// Some clients may want to know if a particular text message or media
+    /// caption contains only emojis so that they can render them bigger for
+    /// added effect.
+    ///
+    /// This function provides that feature with the following
+    /// behavior/limitations:
+    /// - ignores leading and trailing white spaces
+    /// - fails texts bigger than 5 graphemes for performance reasons
+    /// - checks the body only for [`MessageType::Text`]
+    /// - only checks the caption for [`MessageType::Audio`],
+    ///   [`MessageType::File`], [`MessageType::Image`], and
+    ///   [`MessageType::Video`] if present
+    /// - all other message types will not match
+    ///
+    /// See `test_emoji_detection` for examples.
+    pub fn contains_only_emojis(&self) -> bool {
+        let body = match self.content() {
+            TimelineItemContent::Message(msg) => match msg.msgtype() {
+                MessageType::Text(text) => Some(text.body.as_str()),
+                MessageType::Audio(audio) => audio.caption(),
+                MessageType::File(file) => file.caption(),
+                MessageType::Image(image) => image.caption(),
+                MessageType::Video(video) => video.caption(),
+                _ => None,
+            },
+            TimelineItemContent::RedactedMessage
+            | TimelineItemContent::Sticker(_)
+            | TimelineItemContent::UnableToDecrypt(_)
+            | TimelineItemContent::MembershipChange(_)
+            | TimelineItemContent::ProfileChange(_)
+            | TimelineItemContent::OtherState(_)
+            | TimelineItemContent::FailedToParseMessageLike { .. }
+            | TimelineItemContent::FailedToParseState { .. }
+            | TimelineItemContent::Poll(_)
+            | TimelineItemContent::CallInvite
+            | TimelineItemContent::CallNotify => None,
+        };
+
+        if let Some(body) = body {
+            // Collect the graphemes after trimming white spaces.
+            let graphemes = body.trim().graphemes(true).collect::<Vec<&str>>();
+
+            // Limit the check to 5 graphemes for performance and security
+            // reasons
```

Comment:
Performance aspect: why not, I can see.
Security aspect: why? Can you explain a bit please? A link to something maybe?